decode.go

```diff
@@ -336,7 +336,12 @@ func (d *Decoder) decodeValue(dst reflect.Value, src ast.Node) error {
 	}
 	valueType := dst.Type()
 	if unmarshaler, ok := dst.Addr().Interface().(BytesUnmarshaler); ok {
-		b := fmt.Sprintf("%v", src)
+		var b string
+		if scalar, isScalar := src.(ast.ScalarNode); isScalar {
+			b = scalar.GetValue().(string)
```

Comment:
`scalar.GetValue()` will return not only `string` type, but also `int64` or `uint64` and so on .
I think better to use `fmt.Sprint(scalar.GetValue())` to convert to string value .
( also, would you please add test case for integer type ? )

Reply:
OK, I did it.